libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -881,6 +930,70 @@ string YulUtilFunctions::readFromStorageDynamic(Type const& _type, bool _splitFu
 	});
 }
 
+string YulUtilFunctions::readFromMemory(Type const& _type, bool _fromCalldata, bool _padToWords)
+{
+	string functionName =
+		string("read_from_") +
+		(_fromCalldata ? "calldata" : "memory") +
+		"_dynamic_" +
+		_type.identifier();
+
+	return m_functionCollector->createFunction(functionName, [&] {
+		if (auto arrayType = dynamic_cast<ArrayType const*>(&_type))
+		{
+			solAssert(!arrayType->isDynamicallySized(), "");
+			solAssert(!_fromCalldata, "");
+			solAssert(_padToWords, "");
+
+			return Whiskers(R"(
+				function <functionName>(addr) -> value {
+					value := addr
```

Comment:
```suggestion
					value := mload(addr)
```

Reply:
Hmm the old code basically did nothing for this case? https://github.com/ethereum/solidity/blob/develop/libsolidity/codegen/CompilerUtils.cpp#L167-L173

Reply:
I think because there, we already had it on the stack.